tests/virt_autotest/sriov_pci_passthrough.pm

```diff
@@ -0,0 +1,381 @@
+# SUSE's openQA tests
```

Comment:
I  think there is another question we need to think about. You  know sriov is a feature not limited to a single kind of device which can be network card , GPU card or other IO devices. I think your current work only involves network card sriov. So have you ever considered the following questions:
1) Have you taken multiple network cards reside on the same host into consideration ?
2) Have you taken vm guest with multiple srioved device attached from different physical network cards into consideration ?
3) Have you taken vm guest with attached srioved device from different kinds of io devices into consideration ?
4) Can this module and its content be easily reused by other sriov devices ?
So is it realistic that this module is to be splitted into two, namely sriov_pci_passthrough_base.pm and sr_iov_passthrough_network_card.pm ?

Reply:
Good questions.
 - *1: I have taken multiple network devices from the host into consideration. The test will list all the Ethernet devices with SR-IOV functions, locate their drivers and turn on vf functions for them. But no test envionment for verification yet. In the case of my current test verification, there is a SR-IOV ethernet device with 2 ports, you can look them as 2 network cards but with same driver.

 - *2: This test works with multiple SR-IOV ethernet devices passed through from host. Since vfs of all the SR-IOV ethernet devices has been listed(stated in *1), two(or more if needed) random vfs are hot plugged to vm. 

 - *3: This test only involve network device pass through, and it would work even if there would be other SR-IOV devices passed through to guests. It tells the nic from other devices based on the <inerface> and MAC address in vm xml. Still no verification for this case.

 - *4: I have not thought about it but just thought it over. This test is ethernet specific, it is hard to reused by other devices-involved tests. It locates device mostly by device name, MAC address, interface and ect, which is quite different with other devices.